src/sortStudents.ts

```diff
@@ -1,16 +1,69 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+type SortStudentsFunction =
+  (students: Student[], sortBy: SortType, order: SortOrder) => Student[];
+type CallBack = (grades1: number[], grades2: number[], order: string) => number;
 
+const SortCallbackByGrades:CallBack = (grades1, grades2, order) => {
+  const firstAvgGrade = grades1.reduce((p, c) => p + c, 0)
+    / grades1.length;
+  const secondAvgGrade = grades2.reduce((p, c) => p + c, 0)
+    / grades2.length;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
-}
+  return order === 'asc'
+    ? firstAvgGrade - secondAvgGrade
+    : secondAvgGrade - firstAvgGrade;
+};
+
+export const sortStudents: SortStudentsFunction = (students, sortBy, order) => {
+  const arrayCopy = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      return arrayCopy.sort((a: Student, b: Student) => (
+        order === 'asc'
+          ? a.name.localeCompare(b.name)
+          : b.name.localeCompare(a.name)
+      ));
+
+    case SortType.Surname:
+      return arrayCopy.sort((a: Student, b: Student) => (
+        order === 'asc'
+          ? a.surname.localeCompare(b.surname)
+          : b.surname.localeCompare(a.surname)
+      ));
+
+    case SortType.Age:
+      return arrayCopy.sort((a :Student, b: Student) => (order === 'asc'
+        ? a.age - b.age
+        : b.age - a.age));
+
+    case SortType.Married:
+      return arrayCopy.sort((a :Student, b: Student) => (order === 'asc'
+        ? +a.married - (+b.married)
+        : +b.married - (+a.married)));
```

Comment:
Приводь явно до числа
```suggestion
        ? Number(a.married) - (Number(b.married))
```